resources/src/deserialize.rs

```diff
@@ -0,0 +1,93 @@
+use serde::{Deserialize, Deserializer};
+use serde::de;
+use std::str::FromStr;
+
+/// The stellar ammount fields are represented as strings in the horizon api,
+/// however, they look like floats. But, in the actual ledger they are signed
+/// 64-bit integers. This function can be used to convert what we get from
+/// horizon into a signed integer. The precision of the float is assumed to
+/// by out to 7 digits (but if fewer are found it's ok).
+pub(crate) fn amount<'de, D>(d: D) -> Result<i64, D::Error>
+where
+    D: Deserializer<'de>,
+{
+    let s = String::deserialize(d)?;
+    match s.rfind('.') {
+        None => {
+            // There is no decimal so just multiply
+            let parsed_amount =
+                i64::from_str(&s).map_err(|_| de::Error::custom("Failed to parse string field"))?;
+            Ok(parsed_amount * 10_000_000)
+        }
+        Some(decimal_place) => {
+            let number_decimals = s.len() - (decimal_place + 1);
+            if number_decimals > 7 {
+                Err(de::Error::custom(
+                    "Amount has too many digits of precision.",
+                ))
+            } else {
+                let s = s.replace(".", "");
+                let parsed_amount = i64::from_str(&s)
+                    .map_err(|_| de::Error::custom("Failed to parse string field"))?;
```

Comment:
Both arms compute the parsed amount.  Can we do this before the match?

Reply:
Yes, i considered that, but this is actually more efficient. See, on this arm it won't executes after we've removed the decimal point and after we've checked the precision.

Reply:
👍